src/global_styling/variables/_colors.ts

```diff
@@ -26,12 +27,9 @@ import {
 } from '../functions/_colors';
 
 export type _EuiThemeTextColors = {
-  /**
-   * TODO: Allow for `string | keys of LIGHT/DARK` instead of `any`
-   */
-  text: any;
-  title: any;
-  textSubdued: any;
+  text: ColorModeSwitch;
+  title: ColorModeSwitch;
+  textSubdued: ColorModeSwitch;
   textPrimary: string;
```

Comment:
So is `ColorModeSwitch` flexible enough where the value could either be the `LIGHT/DARK` keys OR a simple `string`? Like I could also use it here:
```suggestion
  textSubdued: ColorModeSwitch;
  textPrimary: ColorModeSwitch;
```

Even though the default theme is setting it as a single `string`?

Reply:
You can definitely do `ColorModeSwitch | string`.
Do you think that this will be a common use case? It might be possible to automatically apply the union for primitive types like `string` and `number`.

Reply:
Ah yes, we can certainly do `ColorModeSwitch | string`. But does `ColorModeSwitch` then also strongly type that the value of the LIGHT/DARK keys **also** have to be strings? Maybe that's what you're getting at with the primitive thing?

But yes, I think it'll be most flexible to use this `ColorModeSwitch` for just about every value that is most likely a color . Even if our own themes might use a single value. Basically, I'd end up using this type for ALL the keys in that colors file.

Reply:
To clarify that first question would doing:
`textPrimary: ColorModeSwitch | string;` strongly type to ensure that:

`textPrimary { LIGHT: string; DARK: string }`

Reply:
`ColorModeSwitch` is a generic type that assumes `string` to start, and results in `{ LIGHT: string; DARK: string }`
Or you can do `ColorModeSwitch<number>` to get `{ LIGHT: number; DARK: number }`.
So  `ColorModeSwitch | string` results in `{ LIGHT: string; DARK: string } | string`.

Should be possible to do this automatically, but I'm not sure yet if the more complex examples work, like `ColorModeSwitch<_EuiThemeBaseColors>`. Because the result would be something like
```
{ 
LIGHT: {
  primary: string;
  accent: string;
  success: string;
...
}; 
DARK: {
  primary: string;
  accent: string;
  success: string;
  ... 
}} | { 
  primary: string;
  accent: string;
  success: string;
}